requirements-dev.txt

```diff
@@ -0,0 +1,5 @@
+mock
+pytest
+pytest-cov
+pylint
+flake8
```

Comment:
Should we pin any of the `requirements` or `requirements-dev` packages to specific versions, or just make any changes that come up with updates to things like pylint when they come up?

Reply:
I left it open mostly out of laziness.  I didn't want to prescribe what version of them <had> to be installed just in case somebody else was using earlier versions and didn't really want to take the time right now to find the range of versions that were valid.  If it comes up as an issue later, I'll definitely do this though.